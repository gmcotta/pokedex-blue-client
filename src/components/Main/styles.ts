import styled from 'styled-components';

export const Wrapper = styled.main`
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 4.8rem;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const Heading = styled.h1`
  font-size: 3.2rem;
  font-weight: 700;
`;
