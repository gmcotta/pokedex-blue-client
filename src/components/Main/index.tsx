import * as S from './styles';

const Main = () => (
  <S.Wrapper>
    <S.Heading>Main</S.Heading>
  </S.Wrapper>
);

export default Main;
