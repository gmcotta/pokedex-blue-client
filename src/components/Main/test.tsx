import { render, screen } from '@testing-library/react';
import Main from '.';

describe('<Main />', () => {
  it('should render the component', () => {
    const { container } = render(<Main />);
    expect(screen.getByRole('heading', { name: /main/i })).toBeInTheDocument();
    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' });
    expect(container.firstChild).toMatchSnapshot();
  });
});
