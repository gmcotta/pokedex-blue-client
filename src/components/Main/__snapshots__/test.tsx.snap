// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the component 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 4.8rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 3.2rem;
  font-weight: 700;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Main
  </h1>
</main>
`;
